cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```
@@ -475,6 +482,96 @@ private void updateConfigurationTable() {
   }
 
   /**
+   * The replication Status Tool coprocessors will use replication state table to track
+   * replication status. Create the table here.
+   */
+  private void createReplicationStateTable() throws IOException {
+    LOG.info("Create replicationstate table now.");
+    HBaseTableUtil tableUtil = new HBaseTableUtilFactory(cConf).get();
+    HBaseAdmin admin = new HBaseAdmin(hConf);
+    try {
+      TableId tableId = getReplicationStateTableId(tableUtil);
+      HTableDescriptorBuilder htd = tableUtil.buildHTableDescriptor(tableId);
+      htd.addFamily(new HColumnDescriptor(Constants.ReplicationStatusTool.WRITE_TIME));
+      htd.addFamily(new HColumnDescriptor(Constants.ReplicationStatusTool.REPLICATE_TIME));
+      htd.addFamily(new HColumnDescriptor(Constants.ReplicationStatusTool.SHUTDOWN_TIME));
+      tableUtil.createTableIfNotExists(admin, tableId, htd.build());
+      // Clear shutdown row to keep shutdown time for current instance only
+      saveShutdownTime(0L);
+    } catch (IOException e) {
+      throw Throwables.propagate(e);
```

you don't need to propagate the exception like this. In general, we should avoid using Throwables.propagate whenever possible, as it makes proper error handling very difficult. In this case, the method already throws IOException, so you can remove this catch clause.